rpcserver.go

```diff
@@ -2939,6 +2939,43 @@ func handlePing(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (inter
 	return nil, nil
 }
 
+// getMempoolTxsForAddressRange looks up and returns all transactions from the
+// mempool related to the given address. The, `limit` parameter
+// should be the max number of transactions to be returned. Additionally, if the
+// caller wishes to seek forward in the results some amount, the 'seek'
+// represents how many results to skip.
+// It will return the array of fetched transactions, along with the amount
+// of transactions that were actually skipped.
+func getMempoolTxsForAddressRange(s *rpcServer, addr btcutil.Address, skip int,
+	limit int) ([]*database.TxListReply, int, error) {
+
+	memPoolTxs, err := s.server.txMemPool.FilterTransactionsByAddress(addr)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// If we're asked to skip more transactions than we have,
+	// we skip them all and return an empty slice.
+	if skip >= len(memPoolTxs) {
+		return nil, len(memPoolTxs), nil
+	}
+
+	var result []*database.TxListReply
+
+	// Otherwise, calculate the range we have to return and return it.
+	rangeEnd := skip + limit
+	if rangeEnd > len(memPoolTxs) {
+		rangeEnd = len(memPoolTxs)
+	}
+
+	for _, tx := range memPoolTxs[skip:rangeEnd] {
```

Comment:
I agree ordering the mempool txs would be nice, but II think it should be addressed in a future PR.

In my testing, mempool ordering is not that random, it doesn't change randomly with every call, probably due to the addrindex in mempool being a ```map[string]map[wire.ShaHash]struct{}```, so the order of the txs of a single address doesn't get "disturbed" unless there's a new transaction for that address.

This fix's goal is simply to be able to iterate through all the mempool transactions when paging. For example, if there are 100 mempool txs and I use 10-txs pages, before the fix the client will get the first 10 transactions regardless of the skip value. After the fix, though, the client can successfully retrieve all the 100 transactions as long as there are no new transactions for that address. In my opinion this change alone makes the behavior much better.

Of course, real ordering will be even better. I'll try to have a look at topological sorting after this.